hack/install-git-server.sh

```diff
@@ -17,31 +17,70 @@ set -o nounset
 set -o pipefail
 
 git_server() {
-  echo "Creating Git Server Knative service..."
-  cat << EOF | kubectl apply -f -
-apiVersion: serving.knative.dev/v1
-kind: Service
+  echo "Creating Git Server"
+
+  local name="func-git"
+
+  local namespace
+  namespace="$(kubectl config view --minify --output 'jsonpath={..namespace}')"
+  namespace="${namespace:-"default"}"
+
+  local -r func_git_host="${name}.${namespace}.localtest.me"
+
+  kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
 metadata:
-  name: func-git
+  name: "${name}"
+  namespace: "${namespace}"
   labels:
-    app: git
+    app.kubernetes.io/name: "${name}"
 spec:
-  template:
-    metadata:
-      annotations:
-        autoscaling.knative.dev/max-scale: "1"
-        autoscaling.knative.dev/min-scale: "1"
-        client.knative.dev/user-image: quay.io/mvasek/gitserver
-    spec:
-      containers:
-      - image: quay.io/mvasek/gitserver
-        ports:
+  containers:
+    - name: "${name}"
+      image: quay.io/mvasek/gitserver
+      ports:
         - containerPort: 80
-        resources: {}
-status: {}
+          name: http
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: "${name}"
+  namespace: "${namespace}"
+spec:
+  selector:
+    app.kubernetes.io/name: "${name}"
+  ports:
+    - name: http
+      protocol: TCP
+      port: 80
+      targetPort: http
+  type: ClusterIP
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: "${name}"
+  namespace: "${namespace}"
+spec:
+  ingressClassName: contour-external
+  rules:
+    - host: "${func_git_host}"
+      http:
+        paths:
+          - backend:
+              service:
+                name: "${name}"
+                port:
+                  number: 80
+            pathType: Prefix
+            path: /
```

Comment:
Creating ingress like this does not seem to work in OCP. Some other way would have to be used there.
WDYT @jrangelramos

Reply:
it worked for me on OCP 4.19